src/galileo/handlers/langchain/tool.py

```diff
@@ -0,0 +1,83 @@
+from collections.abc import Sequence
+from typing import Optional
+
+from langchain_core.runnables.base import Runnable
+from langchain_core.tools import BaseTool
+from pydantic import UUID4, BaseModel, ConfigDict, Field
+
+from galileo.constants.protect import TIMEOUT_SECS
+from galileo.protect import invoke
+from galileo_core.schemas.protect.execution_status import ExecutionStatus
+from galileo_core.schemas.protect.payload import Payload as CorePayload
+from galileo_core.schemas.protect.response import Response
+from galileo_core.schemas.protect.ruleset import Ruleset
+
+
+class ProtectToolInputSchema(BaseModel):
+    input: Optional[str] = None
+    output: Optional[str] = None
+
+
+class ProtectTool(BaseTool):
+    name: str = "GalileoProtect"
+    description: str = (
+        "Protect your LLM applications from harmful content using Galileo Protect. "
+        "This tool is a wrapper around Galileo's Protect API, can be used to scan text "
+        "for harmful content, and can be used to trigger actions based on the results."
+        "The tool can be used synchronously or asynchronously, on the input text or output text,"
+        "and can be configured with a set of rulesets to evaluate on."
+    )
+    args_schema: type[BaseModel] = ProtectToolInputSchema
```

Comment:
IIRC this needs to be `pydantic` v1 and can't inherit from `pydantic.BaseModel`.

Reply:
Oh I see.. yea it's `pydantic` v1 in the Protect SDK. Could you share why? I'll add a comment to the line.